builder/amazon/common/step_run_source_instance.go

```diff
@@ -84,24 +84,18 @@ func (s *StepRunSourceInstance) Run(state multistep.StateBag) multistep.StepActi
 	}
 
 	ui.Say("Launching a source AWS instance...")
-	imageResp, err := ec2conn.DescribeImages(&ec2.DescribeImagesInput{
-		ImageIds: []*string{&s.SourceAMI},
-	})
-	if err != nil {
-		state.Put("error", fmt.Errorf("There was a problem with the source AMI: %s", err))
-		return multistep.ActionHalt
-	}
-
-	if len(imageResp.Images) != 1 {
-		state.Put("error", fmt.Errorf("The source AMI '%s' could not be found.", s.SourceAMI))
+	image, ok := state.Get("source_image").(*ec2.Image)
```

Comment:
Yeah, I hit a panic when writing this and realized I should handle the error.

As a new comer to golang seeing panics were a little overwhelming.
My intention here was to provide a fairly easy debug point in the future.

This *shouldn't* ever happen since "we" are the ones putting the object in the shared state. I think I hit this because I had an `ec2.Image` instead of `*ec2.Image`.

It seems like since we have the above, there isn't too much to gain by removing it.
Perhaps we could improve the error message though?

I guess, to phrase it differently,
Yeah, it probably is fine, since this should get hit. If we have it though, is it worth removing?